src/models/edge.ts

```diff
@@ -81,18 +82,20 @@ export enum EdgeType {
   CURVED = 'curved',
 }
 
-export interface IEdge<N extends INodeBase, E extends IEdgeBase> {
-  data: E;
-  position: IEdgePosition;
-  style: IEdgeStyle;
-  state: number;
+export interface IEdge<N extends INodeBase, E extends IEdgeBase> extends ISubject {
   readonly id: any;
   readonly offset: number;
   readonly start: any;
   readonly startNode: INode<N, E>;
   readonly end: any;
```

Comment:
I was considering rewriting the values to use getters for consistency, but some are required in the `IEdgeBase` interface (such as `id`, `start`, `end`), which is also required in `topology.ts`. While I believe it's a good idea to use getters for consistency, it seems unnatural given the existing interface requirements. Is it fine to have a combination of TypeScript getters (e.g., `get x`) and regular getters (e.g., `getX`), or should I consider modifying the API further?